format-common/src/test/java/io/cdap/plugin/format/charset/CharsetTransformingLineRecordReaderTest.java

```diff
@@ -0,0 +1,154 @@
+import org.apache.hadoop.mapreduce.lib.input.CompressedSplitLineReader;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+import static org.powermock.api.mockito.PowerMockito.doAnswer;
+import static org.powermock.api.mockito.PowerMockito.spy;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CharsetTransformingLineRecordReaderTest {
+
+  Configuration conf;
+  FixedLengthCharset fixedLengthCharset;
+  CharsetTransformingLineRecordReader recordReader;
+  FixedLengthCharsetTransformingCodec codec;
+  ByteArrayInputStream inputStream;
+  int availableBytes;
+  FixedLengthCharsetTransformingDecompressorStream decompressorStream;
+  SplitCompressionInputStream compressionInputStream;
+
+  //We set up the input so each line is 5 characters long, which is 20 bytes in UTF-32.
+  final String input = "abcd\nedfg\nijkl\n";
+
+  @Before
+  public void before() throws IOException {
+    //Set up the Compressed Split Line Reader with a buffer size of 4096 bytes.
+    //This ensures the buffer will consume all characters in the input stream if we allow it to.
+    conf = new Configuration();
+    conf.setInt("io.file.buffer.size", 4096);
+
+    fixedLengthCharset = FixedLengthCharset.UTF_32;
+
+    codec = new FixedLengthCharsetTransformingCodec(fixedLengthCharset);
+    codec.setConf(conf);
+
+    inputStream = new ByteArrayInputStream(input.getBytes(fixedLengthCharset.getCharset()));
+    availableBytes = inputStream.available();
+  }
+
+  public void setUpRecordReaderForTest(SplitCompressionInputStream splitCompressionInputStream) throws IOException {
+
+    // Set up record reader to assume we'll read the file from the beggining, and the partition size is 32 bytes
+    // which is 8 characters in UTF-32, meaning we expect to read the first 2 lines for this partition.
+    recordReader = spy(new CharsetTransformingLineRecordReader(
+      fixedLengthCharset,
+      null,
+      new CompressedSplitLineReader(splitCompressionInputStream, conf, null),
+      0,
+      0,
+      32,
+      4096
+    ));
+
+    //We will calculate position based on the number of bytes consumed from the input stream.
+    doAnswer(a -> (long) availableBytes - inputStream.available()).when(recordReader).getFilePosition();
+  }
+
+  @Test
+  public void testGetNextLine() throws IOException {
+    decompressorStream =
+      new FixedLengthCharsetTransformingDecompressorStream(inputStream, FixedLengthCharset.UTF_32, 0, 32);
+    compressionInputStream = new TransformingCompressionInputStream(decompressorStream, 0, 32);
+
+    //Set up test
+    setUpRecordReaderForTest(compressionInputStream);
+
+    //Ensure the first line is read
+    Assert.assertTrue(recordReader.nextKeyValue());
+    Assert.assertEquals(recordReader.getCurrentKey().get(), 0);
```

Comment:
Almost forgot. In this version of JUnit you do `assertEquals(expected, actual)`. Please fix, otherwise errors are looking weird.

Reply:
Good catch. I'll fix that.